pkg/controller/mutationtemplate/mutationtemplate_controller.go

```diff
@@ -124,30 +127,104 @@ func (r *ReconcileMutationTemplate) Reconcile(request reconcile.Request) (reconc
 		if errors.IsNotFound(err) {
 			// Object not found, return.  Created objects are automatically garbage collected.
 			// For additional cleanup logic use finalizers.
+			log.Info("didn't fine the instance - finalizer logic here?")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		log.Info("some other error occurred in getting the instance, requeuing")
+		return reconcile.Result{Requeue: true}, err
 	}
 	log.Info("Reconciling", "MutationTemplate", instance)
 
 	crd, err := r.opa.CreateMutationCRD(context.Background(), instance)
```

Comment:
Don't we need to prevent this from being called if the CRD already exists? Note that in the last checkin, we _assumed_ that it didn't already exist, but we're not making that assumption anymore.